NitroxServer/Communication/Packets/Processors/EntityMetadataUpdateProcessor.cs

```diff
@@ -1,49 +1,76 @@
-﻿using NitroxModel.DataStructures.GameLogic;
-using NitroxModel.DataStructures.Util;
+using NitroxModel.DataStructures.GameLogic;
+using NitroxModel.DataStructures.GameLogic.Entities.Metadata;
 using NitroxModel.Packets;
 using NitroxServer.Communication.Packets.Processors.Abstract;
 using NitroxServer.GameLogic;
 using NitroxServer.GameLogic.Entities;
 
-namespace NitroxServer.Communication.Packets.Processors
+namespace NitroxServer.Communication.Packets.Processors;
+
+public class EntityMetadataUpdateProcessor : AuthenticatedPacketProcessor<EntityMetadataUpdate>
 {
-    public class EntityMetadataUpdateProcessor : AuthenticatedPacketProcessor<EntityMetadataUpdate>
+    private readonly PlayerManager playerManager;
+    private readonly EntityRegistry entityRegistry;
+
+    public EntityMetadataUpdateProcessor(PlayerManager playerManager, EntityRegistry entityRegistry)
     {
-        private readonly PlayerManager playerManager;
-        private readonly EntityRegistry entityRegistry;
+        this.playerManager = playerManager;
+        this.entityRegistry = entityRegistry;
+    }
 
-        public EntityMetadataUpdateProcessor(PlayerManager playerManager, EntityRegistry entityRegistry)
+    public override void Process(EntityMetadataUpdate packet, Player sendingPlayer)
+    {
+        if (!entityRegistry.TryGetEntityById(packet.Id, out Entity entity))
         {
-            this.playerManager = playerManager;
-            this.entityRegistry = entityRegistry;
+            Log.Error($"Entity metadata {packet.NewValue.GetType()} updated on an entity unknown to the server {packet.Id}");
+            return;
         }
 
-        public override void Process(EntityMetadataUpdate packet, Player sendingPlayer)
+        if (TryProcessMetadata(sendingPlayer, entity, packet.NewValue))
         {
-            Optional<Entity> entity = entityRegistry.GetEntityById(packet.Id);
+            entity.Metadata = packet.NewValue;
+            SendUpdateToVisiblePlayers(packet, sendingPlayer, entity);
```

Comment:
Does only sending to visible players here mean anyone outside of view doesn't get the updates and so will be out of sync?

Reply:
Thought this was the case just wanted to check as haven't touched this part of the code base in a while